OrbitClientModel/CaptureDeserializerTest.cpp

```diff
@@ -0,0 +1,408 @@
+      .Times(3)
+      .WillOnce(
+          DoAll(SaveArg<0>(&actual_callstack_event_1_1), InvokeWithoutArgs(check_hash_present_1)))
+      .WillOnce(
+          DoAll(SaveArg<0>(&actual_callstack_event_1_2), InvokeWithoutArgs(check_hash_present_1_2)))
+      .WillOnce(
+          DoAll(SaveArg<0>(&actual_callstack_event_2), InvokeWithoutArgs(check_hash_present_2)));
+
+  EXPECT_CALL(listener, OnCaptureStarted).Times(1);
+  EXPECT_CALL(listener, OnCaptureComplete).Times(1);
+
+  capture_deserializer::internal::LoadCaptureInfo(capture_info, &listener, &empty_stream,
+                                                  &cancellation_requested);
+  EXPECT_TRUE(hash_added_1);
+  EXPECT_TRUE(hash_added_2);
+  EXPECT_TRUE(hash_present_1_1);
+  EXPECT_TRUE(hash_present_1_2);
+  EXPECT_TRUE(hash_present_2);
+
+  EXPECT_EQ(callstack_event_1_1->time(), actual_callstack_event_1_1.time());
+  EXPECT_EQ(callstack_event_1_2->time(), actual_callstack_event_1_2.time());
+  EXPECT_EQ(callstack_event_2->time(), actual_callstack_event_2.time());
+}
+
+TEST(CaptureDeserializer, LoadCaptureInfoTimers) {
+  MockCaptureListener listener;
+  std::atomic<bool> cancellation_requested = false;
+  CaptureInfo empty_capture_info;
+  ErrorMessage actual_error;
+  EXPECT_CALL(listener, OnCaptureStarted).Times(1);
+  EXPECT_CALL(listener, OnCaptureComplete).Times(1);
+  std::stringstream stream;
+
+  TimerInfo timer_1;
+  timer_1.set_start(0);
+  timer_1.set_end(1);
+  timer_1.set_process_id(42);
+
+  std::string serialized_timer_1;
+  timer_1.SerializeToString(&serialized_timer_1);
+  int32_t size_of_timer_1 = serialized_timer_1.size();
+  stream << std::string(absl::bit_cast<char*>(&size_of_timer_1), sizeof(size_of_timer_1))
+         << serialized_timer_1;
+
+  TimerInfo timer_2;
+  timer_2.set_start(3);
+  timer_2.set_end(5);
+  timer_2.set_process_id(2);
+
+  std::string serialized_timer_2;
+  timer_2.SerializeToString(&serialized_timer_2);
+  int32_t size_of_timer_2 = serialized_timer_2.size();
+  stream << std::string(absl::bit_cast<char*>(&size_of_timer_2), sizeof(size_of_timer_2))
+         << serialized_timer_2;
+
+  TimerInfo actual_timer_1;
+  TimerInfo actual_timer_2;
+  EXPECT_CALL(listener, OnTimer)
+      .Times(2)
+      .WillOnce(SaveArg<0>(&actual_timer_1))
+      .WillOnce(SaveArg<0>(&actual_timer_2));
+
+  google::protobuf::io::IstreamInputStream input_stream(&stream);
+  google::protobuf::io::CodedInputStream coded_input(&input_stream);
+
+  capture_deserializer::internal::LoadCaptureInfo(empty_capture_info, &listener, &coded_input,
+                                                  &cancellation_requested);
+
+  EXPECT_EQ(timer_1.start(), actual_timer_1.start());
+  EXPECT_EQ(timer_2.start(), actual_timer_2.start());
+  EXPECT_EQ(timer_1.end(), actual_timer_1.end());
+  EXPECT_EQ(timer_2.end(), actual_timer_2.end());
+  EXPECT_EQ(timer_1.process_id(), actual_timer_1.process_id());
+  EXPECT_EQ(timer_2.process_id(), actual_timer_2.process_id());
```

Comment:
And another change for `operator=`

Reply:
As above, they are generated by a proto.